test/peep_test.exs

```diff
@@ -119,4 +119,23 @@ defmodule PeepTest do
       assert String.contains?(logs, "Dropping #{inspect(event_name)}")
     end
   end
+
+  test "Handlers are detached on shutdown" do
+    metric = Metrics.counter("peep.counter", event_name: [:counter], tags: [])
+
+    {:ok, options} =
+      [
+        name: :"#{__MODULE__}_shutdown_test",
+        metrics: [metric]
+      ]
+      |> Peep.Options.validate()
+
+    {:ok, pid} = GenServer.start(Peep, options, name: options.name)
+
+    assert length(:telemetry.list_handlers([])) == 1
```

Comment:
I'm worried this could fail when tests are run in parallel.

I propose two options:

1. Get the handler id using `:sys.get_state(pid)`, and assert that there is a handler with that id in `:telemetry.list_handlers([])`
2. Create a globally unique metric name, and assert that `:telemetry.list_handlers(MetricName) == 1` when the Peep process is running and `== 0` when the Peep process is terminated.